net-p2p/bitcoin-core/bitcoin-core-28.0_rc1.ebuild

```diff
@@ -0,0 +1,423 @@
+			You are enabling neither USE="daemon" nor USE="gui". This is a valid
+			configuration, but you will be unable to run a Bitcoin node using this
+			installation.
+		EOF
+	fi
+	if use daemon && ! use cli && ! has_version "${CATEGORY}/${PN}[daemon,-bitcoin-cli(-),-cli(-)]" ; then
+		efmt ewarn <<-EOF
+			You are enabling USE="daemon" but not USE="cli". This is a valid
+			configuration, but you will be unable to interact with your bitcoind node
+			via the command line using this installation.
+		EOF
+	fi
+	if ! use berkdb && ! use sqlite &&
+		{ { use daemon && ! has_version "${CATEGORY}/${PN}[daemon,-berkdb,-sqlite]" ; } ||
+		  { use gui && ! has_version "${CATEGORY}/${PN}[gui,-berkdb,-sqlite]" ; } ; }
+	then
+		efmt ewarn <<-EOF
+			You are enabling neither USE="berkdb" nor USE="sqlite". This is a valid
+			configuration, but your Bitcoin node will be unable to open any wallets.
+		EOF
+	fi
+
+	# test/functional/feature_pruning.py requires 4 GB disk space
+	# test/functional/wallet_pruning.py requires 1.3 GB disk space
+	use test && CHECKREQS_DISK_BUILD="6G" check-reqs_pkg_pretend
+}
+
+pkg_setup() {
+	if use test ; then
+		CHECKREQS_DISK_BUILD="6G" check-reqs_pkg_setup
+		python-any-r1_pkg_setup
+	fi
+}
+
+src_prepare() {
+	default
+	! use system-leveldb || rm -r src/leveldb || die
+	if use system-libsecp256k1 ; then
+		rm -r src/secp256k1 || die
+		sed -e '/^DIST_SUBDIRS *=/s/\bsecp256k1\b//' -i src/Makefile.am || die
+	else
+		pushd src/secp256k1 >/dev/null || die
+		AT_NOELIBTOOLIZE=yes eautoreconf
+		popd >/dev/null || die
+	fi
+	eautoreconf
+
+	# we say --disable-util-util, so we can't test bitcoin-util
+	sed -ne '/^  {/{h;:0;n;H;/^  }/!b0;g;\|"exec": *"\./bitcoin-util"|d};p' \
+		-i test/util/data/bitcoin-util-test.json || die
+
+	sed -e 's/^\(complete -F _bitcoind\b\).*$/\1'"$(usev daemon ' bitcoind')$(usev gui ' bitcoin-qt')/" \
+		-i contrib/completions/bash/bitcoind.bash || die
+}
+
+src_configure() {
+	local wallet ; if use berkdb || use sqlite ; then wallet=enable ; else wallet=disable ; fi
+	local myeconfargs=(
+		--disable-static
+		--${wallet}-wallet
+		$(use_with sqlite)
+		$(use_with berkdb bdb)
+		$(use_enable systemtap usdt)
+		$(use_with upnp miniupnpc)
+		$(use_with nat-pmp natpmp)
+		$(use_enable test tests)
+		--disable-bench
+		--disable-fuzz{,-binary}
+		$(use_with qrcode qrencode)
+		--disable-ccache
+		$(use_enable asm)
+		$(use_enable zeromq zmq)
+		$(use_enable man)
+		$(use_enable external-signer)
+		--with-utils
+		$(use_enable {,util-}cli)
+		--enable-util-tx
+		--${wallet}-util-wallet
+		--disable-util-util
+		$(use_with daemon)
+		$(use_with gui gui qt5)
+		$(use_with dbus qtdbus)
+		$(use_with system-leveldb)
+		$(use_with system-libsecp256k1)
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_compile() {
+	default
+
+	if use daemon && ! tc-is-cross-compiler ; then
+		TOPDIR="${S}" bash contrib/devtools/gen-bitcoin-conf.sh || die
+	fi
+	sed -e 's/ To use, copy this file$//p;Tp;:0;n;/save the file\.$/!b0;d;:p;p' \
```

Comment:
is there a reason this can't be in `src_prepare`? (is it a generated file from a template?)

Reply:
It requires executing the just-built `bitcoind` binary, which is why it also can't be run when cross-compiling.

Reply:
I'm talking about the `sed`, sorry.

Reply:
That `sed` command operates on the `share/examples/bitcoin.conf`. file that's (re)generated by `contrib/devtools/gen-bitcoin-conf.sh`. (If we can't regenerate it due to not having built `bitcoind` or having built it for a different architecture, then we just work with the (possibly stale) shipped copy of `share/examples/bitcoin.conf`.)

Reply:
You can see the shipped copy of that file [here](https://github.com/bitcoin/bitcoin/blob/v28.0rc1/share/examples/bitcoin.conf).

The `sed` command changes this:
```
## All possible configuration options are provided. To use, copy this file
## to your data directory (default or specified by -datadir), uncomment
## options you would like to change, and save the file.
```
into this:
```
## All possible configuration options are provided.
```

The reason for making that edit is that we install the file into `/etc/bitcoin/bitcoin.conf`, where it is intended to be in effect, so we don't want to advise the user to copy it in order "to use" it.